id_infrastructure/firestore_uuid_table.py

```diff
@@ -242,10 +245,32 @@ def get_all_mappings(self):
         :rtype: dict
         """
         self._mappings_cache = {}
-        for mapping in self._client.collection(f"tables/{self._table_name}/mappings").get():
+        for mapping in self._mappings_ref().get():
             self._mappings_cache[mapping.id] = mapping.get(_UUID_KEY_NAME)
         return self._mappings_cache.copy()
 
     @staticmethod
     def generate_new_uuid(prefix):
         return prefix + str(uuid.uuid4())
+
+    def delete_mappings(self, mappings, regexp):
+        """
+        Deletes the mappings specified.
+
+        :param mappings: Dictionary of data -> uuid
+        :type mappings: dict of str -> str
+        :param regexp: Regular expression object.
+        :type regexp: re.Pattern
```

Comment:
I don't think passing in a regex is the right thing to do here, as it seems to me to be a layering violation - at the database api we should only be thinking about the mappings to be deleted. I think it's better to pass in the (data <-> uuid) mappings and check that the data and uuids match the database before proceeding.

So on L273 instead of re-running the regex, you'd say something like
assert mappings[mapping_id] == mapping.get("uuid")
(not sure if that's the correct code, but hopefully you get the idea)

Reply:
Updated, thanks.